service/dap/server_test.go

```diff
@@ -471,7 +524,463 @@ func TestStackTraceRequest(t *testing.T) {
+					}
+					expectVarExact(t, locals, 19, "chnil", "nil <chan int>", 0)
+					// reflect.Kind == Func - see testvariables
+					// reflect.Kind == Interface
+					expectVarExact(t, locals, 38, "ifacenil", "nil <interface {}>", 0)
+					expectVarExact(t, locals, 35, "iface2", "<interface {}>", 1019)
+					{
+						client.VariablesRequest(1019)
+						iface2 := client.ExpectVariablesResponse(t)
+						expectVars(t, iface2, "iface2", 1)
+						expectVarExact(t, iface2, 0, "data", `"test"`, 0)
+					}
+					expectVarExact(t, locals, 36, "iface3", "<interface {}>", 1020)
+					{
+						client.VariablesRequest(1020)
+						iface3 := client.ExpectVariablesResponse(t)
+						expectVars(t, iface3, "iface3", 1)
+						expectVarExact(t, iface3, 0, "data", "<map[string]go/constant.Value> (length: 0)", 1064)
+						{
+							client.VariablesRequest(1064)
+							data := client.ExpectVariablesResponse(t)
+							expectVars(t, data, "data", 0)
+						}
+					}
+					// reflect.Kind == Map
+					expectVarExact(t, locals, 21, "mnil", "nil <map[string]main.astruct>", 0)
+					expectVarExact(t, locals, 22, "m2", "<map[int]*main.astruct> (length: 1)", 1012)
+					{
+						client.VariablesRequest(1012)
+						m2 := client.ExpectVariablesResponse(t)
+						expectVars(t, m2, "m2", 1)
+						expectVarRegex(t, m2, 0, "1", "\\<\\*main\\.astruct\\>\\(0x[0-9a-f]+\\)", 1065)
+						{
+							client.VariablesRequest(1065)
+							m2_1 := client.ExpectVariablesResponse(t)
+							expectVars(t, m2_1, "m2[1]", 1)
+							expectVarExact(t, m2_1, 0, "", "<main.astruct>", 1066)
+							{
+								client.VariablesRequest(1066)
+								m2_1val := client.ExpectVariablesResponse(t)
+								expectVars(t, m2_1val, "*m2[1]", 2)
+								expectVarExact(t, m2_1val, 0, "A", "10", 0)
+								expectVarExact(t, m2_1val, 1, "B", "11", 0)
+							}
+						}
+					}
+					expectVarExact(t, locals, 23, "m3", "<map[main.astruct]int> (length: 2)", 1013)
+					{
+						client.VariablesRequest(1013)
+						m3 := client.ExpectVariablesResponse(t)
+						expectVars(t, m3, "m3", 2)
+						expectVarExact(t, m3, 0, "<main.astruct>", "42", 1067)
+						{
+							client.VariablesRequest(1067)
+							m3_0 := client.ExpectVariablesResponse(t)
+							expectVars(t, m3_0, "m3[0]", 2)
+							expectVarExact(t, m3_0, 0, "A", "1", 0)
+							expectVarExact(t, m3_0, 1, "B", "1", 0)
+						}
+						expectVarExact(t, m3, 1, "<main.astruct>", "43", 1068)
+						{
+							client.VariablesRequest(1068)
+							m3_1 := client.ExpectVariablesResponse(t)
+							expectVars(t, m3_1, "m3[1]", 2)
+							expectVarExact(t, m3_1, 0, "A", "2", 0)
+							expectVarExact(t, m3_1, 1, "B", "2", 0)
+						}
+					}
+					expectVarExact(t, locals, 24, "m4", "<map[main.astruct]main.astruct> (length: 2)", 1014)
+					{
+						client.VariablesRequest(1014)
+						m4 := client.ExpectVariablesResponse(t)
+						expectVars(t, m4, "m4", 4)
+						expectVarExact(t, m4, 0, "[key 0]", "<main.astruct>", 1069)
```

Comment:
This is an improvement over what vscode-go does. I noticed that that because they use key as name and value as value, if the key a compound variable, there is no way to expand to view the details because they lose the corresponding variable reference regardless of what the value is. When the value is scalar, we can still print both on the same line and use the expansion option for the key. But when both the value and the key need the expansion we are forced to put them on different lines. I added bogus key/value "names". Let me know if you have a better suggestion.

Before
<img src="https://user-images.githubusercontent.com/51177946/88311587-ede43100-ccc5-11ea-9390-73a4a5353d1a.png" alt="alt text" width="500">
After
<img src="https://user-images.githubusercontent.com/51177946/88311692-12d8a400-ccc6-11ea-8434-de8dfde5ab95.png" alt="alt text" width="420">